oop-object_interactions-medium-q3 (#684)
canvas-gamification/canvas-gamification-junit-tests: src/oop/object_interactions/medium/q3/QuestionText.md

# oop-object_interactions-medium-q3

- Java question

## Newspaper Class

> The `Newspaper` class currently has the following items:

Reviewer: Please add back in the question context

- Attributes:
    - An `int` called **myYear**
    - An `int` called **myMonth**
    - An `int` called **myDay**
    - A `String` called **myStory**
    - None of these attributes should be accessible from other classes.

- Constructors:
    - A constructor with no parameters that initializes all the int attributes to -1 and the **myStory** to an
      empty `String`.
    - It is accessible from other classes.

- Methods:
    - Getter methods for the **myStory**, **myYear**, **myMonth**, and **myDay** attributes that are accessible from other
      classes.

You must add the following items to the class:

- Constructors:

>     - A constructor that takes three `int` and one `String` in that order and initialized the **myYear**, **myMonth**,

Reviewer: ```suggestion
    - A constructor that takes three `int` and one `String` in that order and initializes the **myYear**, **myMonth**,
```

      **myDay**, and **myStory** attribute in that order.
    - It should be accessible from other classes.

- Methods:

>     - A `makeCopy` method that take another **Newspaper** object and change the attribute of the object to the attributes

Reviewer: I think the description is currently not clear on which object you are modifying the attributes of. Please clarify the wording, and also specify the return type of the method.

      of the input parameter **Newspaper** object.

>     - A `toString` method which returns the information about the class in the following `String` structure:

Reviewer: I also think this toString description is a bit unclear. Instead of "in the order", maybe say "has the values" and specify that you should form the date with slashes between the format YEAR/MONTH/DAY

      empty `String`.
    - It is accessible from other classes.

- Methods:
    - Getter methods for **myStory**, **myYear**, **myMonth**, and **myDay** attributes that are accessible from other
      classes.

You must add the following items to the class:

- Constructors:
    - A constructor that takes three `int` and one `String` in that order and initialized the **myYear**, **myMonth**,
      **myDay**, and **myStory** attribute in that order.
    - It should be accessible from other classes.

- Methods:
    - A `makeCopy` method that take another **Newspaper** object and change the attribute of the class to the attributes
      of the input parameter **Newspaper** object.
    - A `toString` method which returns the information about the class in the following `String` structure:
      if the **myYear**, **myMonth**, **myDay**, and **myStory** attributes are in order 2023, 6,
      9, and `It's graduation day!`. Then the `String` should be `2023/6/9:The story is: It's graduation day!`.
    - All methods must be accessible from other classes.

## TestNewspaper Class

Your `TestNewspaper` class must have the following items:

- Main Method:

>     - A `main` method which initializes one **Newspaper** class with the date 2021/12/5 and the story of

Reviewer 2: I feel like this should utilize the makeCopy method as well

# oop-object_interactions-medium-q3

- Java question

## Question Text

You work at a printing company and your summer job is to make copies of original works. Make a **Newspaper** class that

> has attributes to store the year, month, and day of the print as well as the story.

Reviewer 2: get rid of duplicate class information

## Newspaper Class

The `Newspaper` class currently has the following items:

- Attributes:
    - An `int` called **myYear**
    - An `int` called **myMonth**
    - An `int` called **myDay**
    - A `String` called **myStory**
    - None of these attributes should be accessible from other classes.

- Constructors:
    - A constructor with no parameters that initializes all the int attributes to -1 and the **myStory** to an
      empty `String`.
    - It is accessible from other classes.

- Methods:

>     - Getter methods for **myStory**, **myYear**, **myMonth**, and **myDay** attributes that are accessible from other

Reviewer 2: ```suggestion
    - Getter methods for the **myStory**, **myYear**, **myMonth**, and **myDay** attributes that are accessible from other
```

      classes.

You must add the following items to the class:

- Constructors:
    - A constructor that takes three `int` and one `String` in that order and initialized the **myYear**, **myMonth**,
      **myDay**, and **myStory** attribute in that order.
    - It should be accessible from other classes.

- Methods:

>     - A `makeCopy` method that take another **Newspaper** object and change the attribute of the class to the attributes

Reviewer 2: I'm unsure that this should say "class", it doesn't feel like the proper terminology but I may be wrong

Author: I think when talking about instances we use the "object" term. Like we don't say that arrays are classes.

      of the input parameter **Newspaper** object.
    - A `toString` method which returns the information about the class in the following `String` structure:
      if the **myYear**, **myMonth**, **myDay**, and **myStory** attributes are in order 2023, 6,
      9, and `It's graduation day!`. Then the `String` should be `2023/6/9:The story is: It's graduation day!`.
    - All methods must be accessible from other classes.

## TestNewspaper Class

Your `TestNewspaper` class must have the following items:

- Main Method:
    - A `main` method which initializes one **Newspaper** class with the date 2021/12/5 and the story of
      `The votes are in!` and print it using **toString** method.

>     - Then initialize a new **Newspaper** object without parameters, and print it, and use the `makeCopy` method to copy the previous

Reviewer 2: ```suggestion
    - Then initialize a new **Newspaper** object without parameters, print it, and use the `makeCopy` method to copy the previous
```

# oop-object_interactions-medium-q3

- Java question

## Question Text

> You work at a printing company and your summer job is to make copies of original works. Make a **Newspaper** class with

Reviewer: Please also mention they need a test class.

> You work at a printing company and your summer job is to make copies of original works. Make a **Newspaper** class with
> the following description and **TestNewspaper** for testing the **Newspaper** class.

Reviewer: ```suggestion
You work at a printing company and your summer job is to make copies of original works. Create a **Newspaper** class with
the following description to model a newspaper you copied.
You must also write a **TestNewspaper** class to test your **Newspaper** class.
```

## Newspaper Class

The `Newspaper` class currently has the following items:

- Attributes:
    - An `int` called **myYear**
    - An `int` called **myMonth**
    - An `int` called **myDay**
    - A `String` called **myStory**
    - None of these attributes should be accessible from other classes.

- Constructors:
    - A constructor with no parameters that initializes all the int attributes to -1 and the **myStory** to an
      empty `String`.
    - It is accessible from other classes.

- Methods:
    - Getter methods for the **myStory**, **myYear**, **myMonth**, and **myDay** attributes that are accessible from
      other
      classes.

You must add the following items to the class:

- Constructors:
    - A constructor that takes three `int` and one `String` in that order and initializes the **myYear**, **myMonth**,
      **myDay**, and **myStory** attribute in that order.
    - It should be accessible from other classes.

- Methods:

>     - A `makeCopy` method that take another **Newspaper** object and change the attribute of the object that calls the
>       method to the attributes of the input parameter **Newspaper** object in input. This method must not return any
>       value.

Reviewer: ```suggestion
    - A `makeCopy` method that takes another **Newspaper** object and changes the attribute of the object that called the
      method to the attributes of the input  **Newspaper** object. This method must not return any
      value.
```
